src/main/java/net/minecraftforge/common/crafting/RecipeType.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.crafting;
+
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.ResourceLocation;
+
+public class RecipeType<T extends IRecipe>
+{
+    protected final ResourceLocation id;
+    protected final Class<T> baseClass;
+    
+    public RecipeType(ResourceLocation id, Class<T> baseClass) {
+        this.id = id;
+        this.baseClass = baseClass;
+    }
+
+    public ResourceLocation getId()
+    {
+        return id;
+    }
+
+    /**
+     * @return The class that all recipes of this type must extend.
+     */
+    public Class<T> getBaseClass()
+    {
+        return baseClass;
+    }
```

Comment:
Is there a method for modders to share recipe types?
An enum system with our extender or some form of get(ResourceLocation) would be good.